src/main/java/com/exadel/discountwebapp/category/controller/CategoryController.java

```diff
@@ -0,0 +1,42 @@
+package com.exadel.discountwebapp.category.controller;
+
+import com.exadel.discountwebapp.category.service.CategoryService;
+import com.exadel.discountwebapp.category.vo.RequestCategoryVO;
+import com.exadel.discountwebapp.category.vo.ResponseCategoryVO;
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/category")
+@AllArgsConstructor
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    @GetMapping
+    public List<ResponseCategoryVO> findAll() {
+        return categoryService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public ResponseCategoryVO findById(@PathVariable long id) {
+        return categoryService.findById(id);
+    }
+
+    @PostMapping
+    public ResponseCategoryVO create(@RequestBody RequestCategoryVO request) {
```

Comment:
Let's have the following VO naming convention:
CategoryBaseVO
CategoryRequestVO
CategoryResponseVO

Reply:
done